cmake/sbom.cmake

```diff
@@ -317,7 +317,7 @@ macro(sbom_find_python)
 	endif()
 
 	if(Python3_FOUND)
-		if(NOT DEFINED SBOM_HAVE_PYTHON_DEPS)
+		if(NOT SBOM_HAVE_PYTHON_DEPS)
```

Comment:
In your proposed fix, python checks the dependencies at every invocation of this macro, in case it could not find it before. In the original code, it only checks it once, regardless whether the packages were found or not. The latter is more efficient, but it has the drawback that if you would install the missing dependencies on your system, you have to clear the CMake cache to let it check again. Is that the behavior you are trying to fix?

Reply:
My envrionment python modules are not enougth, but variable `SBOM_HAVE_PYTHON_DEPS` is "". `if(NOT DEFINED SBOM_HAVE_PYTHON_DEPS)` cannot cover, but `if("${ARGN}" STREQUAL "REQUIRED" AND NOT SBOM_HAVE_PYTHON_DEPS)`  will.  Then python module check cannot work.  I think it's better to judge the variable with same rule.  `if(NOT SBOM_HAVE_PYTHON_DEPS)` is better, it contains `if(NOT DEFINED SBOM_HAVE_PYTHON_DEPS)`

Reply:
I didn't find files from dist folder......  I just clone and try build with cmake, debug with the message from `fatal error`. I get it now after I read build.sh

Reply:
I cannot reproduce your issue with this variable, so I'll leave it as is.